openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ReturnTypeGenerator.java

```diff
@@ -272,39 +315,47 @@ private ReturnTypeDescriptorNode getReturnNodeForSchemaType(Set<Map.Entry<String
     /**
      * Generate union type node when operation has multiple responses.
      */
-    private TypeDescriptorNode handleMultipleResponse(ApiResponses responses) throws BallerinaOpenApiException {
+    private TypeDescriptorNode handleMultipleResponse(ApiResponses responses, List<String> returnDescription)
+            throws BallerinaOpenApiException {
 
         Set<String> qualifiedNodes = new LinkedHashSet<>();
 
         for (Map.Entry<String, ApiResponse> response : responses.entrySet()) {
             String responseCode = response.getKey().trim();
             String code = GeneratorConstants.HTTP_CODES_DES.get(responseCode);
             Content content = response.getValue().getContent();
+            String typeName = null;
 
             if (code == null && !responseCode.equals(GeneratorConstants.DEFAULT)) {
                 throw new BallerinaOpenApiException(String.format(OAS_SERVICE_107.getDescription(), responseCode));
             }
 
             if (responseCode.equals(GeneratorConstants.DEFAULT)) {
                 TypeDescriptorNode record = createSimpleNameReferenceNode(createIdentifierToken(HTTP_RESPONSE));
-                qualifiedNodes.add(record.toSourceCode());
+                typeName = record.toSourceCode();
             } else if (content == null && response.getValue().get$ref() == null ||
                     content != null && content.size() == 0) {
                 //key and value
                 QualifiedNameReferenceNode node = GeneratorUtils.getQualifiedNameReferenceNode(GeneratorConstants.HTTP,
                         code);
-                qualifiedNodes.add(node.toSourceCode());
+                typeName = node.toSourceCode();
             } else if (content != null) {
                 TypeDescriptorNode bodyType = handleMultipleContents(content.entrySet());
                 //Check the default behaviour for return type according to POST method.
                 boolean isWithOutStatusCode =
                         (httpMethod.equals(POST) && responseCode.equals(GeneratorConstants.HTTP_201)) ||
                                 (!httpMethod.equals(POST) && responseCode.equals(GeneratorConstants.HTTP_200));
                 if (isWithOutStatusCode) {
-                    qualifiedNodes.add(bodyType.toSourceCode());
+                    typeName = bodyType.toSourceCode();
                 } else {
                     SimpleNameReferenceNode node = createReturnTypeInclusionRecord(code, bodyType);
-                    qualifiedNodes.add(node.name().text());
+                    typeName = node.name().text();
+                }
+            }
+            if (typeName != null) {
+                qualifiedNodes.add(typeName);
+                if (response.getValue().getDescription() != null && !response.getValue().getDescription().isEmpty()) {
+                    returnDescription.add(typeName.trim() + "|" + response.getValue().getDescription().trim());
```

Comment:
Shall we reuse the constant value for the pipeline "|" if we defined it already?

Reply:
Updated